File: e-commerce/urunKatalog4/ui/src/app/model/urun.ts
export interface Urun{
    urunId: number,
    urunName: string,
    urunDescription: string,
    urunColor: string,
    urunMarka: string,
    urunDurum: string,
    urunImageUrl: string,
    urunFiyat: number,
   isOfferable?: true,
    isApproved?: true,
    stok?: number,
    type?: string

}

export interface UrunParams{
    orderBy:string;
    searchTerm?:string;
    type:string[];
    marka:string[];
    pageNumber:number;
    pageSize:number;
    
}

File: e-commerce/urunKatalog4/backend/Entity/Urun.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace urunKatalog4.Entity
{
    public class Urun
    {
        [Key]
        public int UrunId { get; set; }
        [MaxLength(100)]//ürün adı max 100 karakter olacak.
        public string UrunName { get; set; }
        [MaxLength(500)]//ürün açıklaması max 500 karakter olacak.
        public string UrunDescription { get; set; }
        public string UrunColor { get; set; }
        public string UrunMarka { get; set; }
        public string UrunDurum { get; set; }
        public string UrunImageUrl { get; set; }
        public int UrunFiyat { get; set; }
        public bool isOfferable { get; set; }
        public int Stok { get; set; }
        public string Type { get; set; }
        public string PublicId { get; set; }
    }
}
